src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantValidator.cs

```diff
@@ -84,13 +84,29 @@ public async Task<IEnumerable<ModelError>> ValidateAsync(TenantViewModel model)
                 errors.Add(new ModelError(nameof(model.Name), S["A tenant with the same name already exists."]));
             }
 
-            var allOtherShells = allSettings.Where(t => !String.Equals(t.Name, model.Name, StringComparison.OrdinalIgnoreCase));
+            var allOtherShells = allSettings.Where(tenant => !String.Equals(tenant.Name, model.Name, StringComparison.OrdinalIgnoreCase));
 
             if (allOtherShells.Any(tenant => String.Equals(tenant.RequestUrlPrefix, model.RequestUrlPrefix?.Trim(), StringComparison.OrdinalIgnoreCase) && DoesUrlHostExist(tenant.RequestUrlHost, model.RequestUrlHost)))
             {
                 errors.Add(new ModelError(nameof(model.RequestUrlPrefix), S["A tenant with the same host and prefix already exists."]));
             }
 
+            if (selectedProvider.HasConnectionString)
+            {
+                var allOtherShellsHaveConnectionString = allOtherShells
+                    .Where(tenant => !String.IsNullOrEmpty(tenant.ShellConfiguration?["ConnectionString"]))
+                    .ToList();
+
+                if (allOtherShellsHaveConnectionString.Any() &&
+                    allOtherShellsHaveConnectionString.Any(
+                        tenant => String.Equals(model.ConnectionString, tenant.ShellConfiguration["ConnectionString"], StringComparison.OrdinalIgnoreCase) &&
```

Comment:
I don't think this is a good check. For example, look a the following variation of SQL-Server strings
```
Data Source=localhost;Database=Orchard;Integrated Security=True;
Server=localhost;Database=Orchard;Integrated Security=True;
Server=localhost;Database=Orchard;User Id=username1;Password=password
User Id=username1;Server=localhost;Database=Orchard;Password=password
```

All the above should be treated the same since they point to the same exact thing when looking at the combination of 3 (Provider, Server, and Database) + prefix.

The way to do it here is to let the provider parse the connection string and give you a common `DbConnection` object. Like create `IDbConnectionFactory` that would return a DbConnection from a given screen and then use the common info in the `DbConnection` to create a unique key for each connection and see if these exists.

```

```

Reply:
We already provider a sample for each connection string ;) nothing but I can handle this easliy if we need to support such variations

Reply:
Yeah, I don't think that is enough check. Not everyone follow the sample string and even then, sometimes people use different username to to connect to the same database. 

Do keep in mind that my original PR was just to inspect all connections string in the configuration "after paring them". But during the last meeting, @sebastienros suggest that we take it one step further and actually validation the database connection to make sure the the tables do not physically exists on the server. So if the recommendation not to do connection validation, I can tweak the PR #11822 and eliminate the physical connection validation step and just reply on the connection settings like how the PR original started. But I personally like @sebastienros recommendation which is why I tooks the PR one step further. 

I would suggest we wait and see more feedback from @jtkech , @Skrypt  and @agriffard so we are not wasting time making changes into both placed. Meanwhile I suggest you tag this PR with do not merge.

Reply:
As I said in other PR, let your check the table existance to avoid invalid serial instead of table prefix issue, this way it's clear for all what's going on